src/modm/driver/can/mcp2515_impl.hpp

```diff
@@ -231,79 +299,127 @@ modm::Mcp2515<SPI, CS, INT>::getMessage(can::Message& message)
 
 	// RX0IF or RX1IF respectivly were already cleared automatically by rising CS.
 	// See section 12.4 in datasheet.
-
 	return true;
 }
 
+template <typename SPI, typename CS, typename INT>
+modm::ResumableResult<bool>
+modm::Mcp2515<SPI, CS, INT>::update(){
+	/// todo
+	/// this should be a timer interrupt 
```

Comment:
I think this is better off in a main loop in most cases, so we should leave that decision to the user where this function get called regularly (as often as possible).
The user is then also responsible to take care of locks/mutexes and we don't bloat the driver.

Reply:
That might be your use-case, but I don't see the benefit in enforcing it in this driver.

> That way we not only reduce the code complexity

No, it will actually increase complexity.

> but also keep the modm::can api exactly the same

Why isn't that possible the other way?

Reply:
Because you don't have an update function when using the internal can driver .. when this is done implicitly, then you don't have to explicitly call it

Reply:
Ah, now I understand your intention.
I'm sorry, but that is impossible to implement because there is no cross-platform (think of STM32, SAM, AVR, RPi Pico, Linux/RPi, ...) way to generate a timer interrupts at a specified frequency.

The user has to set up a way that calls `update()` with a (ideally variable) frequency, like the user also has to initialize Spi.

Reply:
hm ok, so the update function will stay and it will be void in the end